odoo/api.py

```diff
@@ -499,29 +499,28 @@ def __new__(cls, cr, uid, context, su=False, uid_origin=None):
         if uid_origin == SUPERUSER_ID:
             uid_origin = None
 
-        assert context is not None
-
         # determine transaction object
         transaction = cr.transaction
         if transaction is None:
             transaction = cr.transaction = Transaction(Registry(cr.dbname))
 
         # if env already exists, return it
         for env in transaction.envs:
-            if (env.cr, env.uid, env.context, env.su, env.uid_origin) == (cr, uid, context, su, uid_origin):
+            if (env.cr, env.uid, env.su, env.uid_origin, env.context) == (cr, uid, su, uid_origin, context):
                 return env
 
         # otherwise create environment, and add it in the set
         assert isinstance(cr, BaseCursor)
         self = object.__new__(cls)
-        self.cr, self.uid, self.context, self.su = self.args = (cr, uid, frozendict(context), su)
-        self.uid_origin = uid_origin
-
-        self.transaction = self.all = transaction
+        self.cr, self.uid, self.su, self.uid_origin = cr, uid, su, uid_origin
+        self.context = frozendict(context)
+        self.transaction = transaction
```

Comment:
I wonder if we shouldn't add a `all` property with a deprecation warning. Not sure it is mandatory but it may help adapting the code if used anywhere and we could remove it in 18.1

Reply:
I looked in all OCA modules, I only found 2 weird usage of `env.all`. I prefer just to remove it, it is too technical IMO.